plugins/dell/fu-plugin-dell.c

```diff
@@ -522,6 +523,115 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static void Esys_Finalize_autoptr_cleanup (ESYS_CONTEXT *esys_context)
+{
+	Esys_Finalize (&esys_context);
+}
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESYS_CONTEXT, Esys_Finalize_autoptr_cleanup)
+
+static gchar *
+fu_plugin_dell_get_tpm_capability (ESYS_CONTEXT *ctx, guint32 query)
+{
+	TSS2_RC rc;
+	guint32 val;
+	g_autofree gchar *result = NULL;
+	g_autofree TPMS_CAPABILITY_DATA *capability = NULL;
+	rc = Esys_GetCapability (ctx, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
+	                         TPM2_CAP_TPM_PROPERTIES, query, 1, NULL, &capability);
+	if (rc != TSS2_RC_SUCCESS) {
+		g_debug ("capability request failed for query %x", query);
+		return NULL;
+	}
+	if (capability->data.tpmProperties.count == 0) {
+		g_debug ("no properties returned for query %x", query);
+		return NULL;
+	}
+	if (capability->data.tpmProperties.tpmProperty[0].property != query) {
+		g_debug ("wrong query returned (got %x expected %x)",
+			 capability->data.tpmProperties.tpmProperty[0].property,
+			 query);
+		return NULL;
+	}
+
+	val = GUINT32_FROM_BE (capability->data.tpmProperties.tpmProperty[0].value);
+	result = g_strstrip (g_strndup ((gchar *)&val, 4));
```

Comment:
I'm not sure I like g_strstrip on the unsanitized chars. Is that safe for non-ASCII?

Reply:
Adjusted it to sanitize characters and replace with spaces before the `g_strstrip` call.  `g_strstrip` specifically looks for space characters when it trims.

Reply:
maybe in a related way we change `fu_common_strstrip` to accept the string length and use that instead?

Reply:
Adjusted to use `fu_common_strstrip` instead.